### PR TITLE
Testing expected behavior of blacklist config

### PR DESCRIPTION
**Pull request by jewzaam**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -7,16 +7,21 @@ TESTTARGETS := $(shell go list -e ./... | egrep -v "/(vendor)/")
...
+all: check build test
+
 .PHONY: check
 check: ## Lint code
 	gofmt -s -l $(shell go list -f '{{ .Dir }}' ./... ) | grep ".*\.go"; if [ "$$?" = "0" ]; then gofmt -s -d $(shell go list -f '{{ .Dir }}' ./... ); exit 1; fi
 	go vet ./cmd/... ./pkg/...
 
 .PHONY: build
-build: test ## Build binary
+build: check ## Build binary
```
Comment (lisa) [inline Makefile]:
```suggestion
build: test check ## Build binary
```

We probably want to test before `build`?

File: pkg/dedicatedadmin/dedicatedadmin_test.go [hunk 2]
```diff
@@ -42,3 +45,65 @@ func TestBlacklist(t *testing.T) {
 		}
 	}
 }
+
+func TestExpectedBlacklist(t *testing.T) {
+	var configfilename = "../../manifests/50-dedicated-admin-operator-config.ConfigMap.yaml"
```
Comment (lisa) [inline pkg/dedicatedadmin/dedicatedadmin_test.go:50]: I'm not sure what the benefit of this is. The unique thing the dedicatedadmin package is doing is splitting a string on `,`. Looping over the resulting slice and calling `regexp.MatchString` is well-tested by upstream Go, thus we don't need to test the loop runs `n` times, or that `regexp.MatchString` is behaving [as documented](https://golang.org/pkg/regexp/#Regexp.MatchString).

We can provide fixture data (to `IsBlackListedNamespace`) that is divorced from the live data and intentionally cause breakages in code that _we're_ responsible for. That is, we can provide, `^[kube-.*` and demonstrate a failure to match `kube-system` (because of the `[`); we would never discover this until a typo was included in the `manifests` config file because this test assumes clean input.

So, then, what _are_ the edge cases? A malformed input string, either because of a failure to correctly split on `,`, or because the regular expression doesn't [compile](https://golang.org/pkg/regexp/#Compile). These cases don't particularly care about the distributed config in `manifests`.

I hope with a smaller changeset this comment makes more sense than in the context of the larger changeset (https://github.com/openshift/dedicated-admin-operator/pull/13).

**Review by lisa — CHANGES_REQUESTED** [review]
Happy to chat on 👖 to go over this more.

**Comment by jewzaam** [comment]
closing in favor of https://github.com/openshift/dedicated-admin-operator/issues/16
